pkg/crc/podman/podman_cache.go

```diff
@@ -0,0 +1,118 @@
+package podman
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/download"
+	"github.com/code-ready/crc/pkg/embed"
+	"github.com/code-ready/crc/pkg/extract"
+	crcos "github.com/code-ready/crc/pkg/os"
+	"github.com/pkg/errors"
+)
+
+// Podman is a struct with methods designed for dealing with the podman binary
+type PodmanCached struct{}
+
+func (podman *PodmanCached) EnsureIsCached() error {
+	if !podman.IsCached() {
+		err := podman.cachePodman()
+		if err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func (podman *PodmanCached) IsCached() bool {
+	if _, err := os.Stat(filepath.Join(constants.CrcBinDir, constants.PodmanBinaryName)); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func (podman *PodmanCached) getPodman(destDir string) (string, error) {
+	logging.Debug("Trying to extract podman from crc binary")
+	archiveName := filepath.Base(constants.GetPodmanUrl())
+	destPath := filepath.Join(destDir, archiveName)
+	err := embed.Extract(archiveName, destPath)
+	if err != nil {
+		logging.Debug("Downloading podman")
+		return download.Download(constants.GetPodmanUrl(), destDir, 0600)
+	}
+
+	return destPath, err
+}
+
+// cachePodman downloads and caches the podman binary into the CRC directory
+func (podman *PodmanCached) cachePodman() error {
+	if podman.IsCached() {
+		return nil
+	}
+
+	// Create tmp dir to download the podman tarball
+	tmpDir, err := ioutil.TempDir("", "crc")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpDir)
+	assetTmpFile, err := podman.getPodman(tmpDir)
+	if err != nil {
+		return err
+	}
+
+	// Extract the tarball and put it the cache directory.
+	err = extract.Uncompress(assetTmpFile, tmpDir)
```

Comment:
Extending UncompressWithFilter to extract only the binary you are interested in was not possible?

Reply:
Given a tarball, UncompressWithFilter will only extract the files which names match the filter (ie the filter returns `'true`)

Reply:
Ah right, following logic take care of it :+1:  
```
	// the target location where the dir/file should be created
		path := filepath.Join(targetDir, header.Name)
		if fileFilter != nil && !fileFilter(path) {
			logging.Debugf("untar: Skipping %s", path)
			continue
		}
```

Reply:
This is why it took so long. I was first trying to modify `UncompressWithFilter` to work for me but then I realized it'll be too much trouble than it's worth.

Reply:
unless there are strong feelings here, I'm resolving this thread.